### ISSUE TITLE
Availability in Build settings and Info.plist instructions work locally on MAC but not on Circle CI

### ISSUE THREAD

**Issue by JanithaR** [react-native-config/react-native-config#409]
The vars are always correctly populated when building/archiving locally on my MAC but always empty when built/archived on Circle CI.

I'm not really sure what other information to give on this other than that. Basically, I ran into this problem when integrating Codepush to my app, where they expect the keys to be defined in the Info.plist file. I took the traditional route of specifying build settings as a temporary workaround for the moment.

Let me know how I can help with any additional logs or similar.

**Comment by luis-f-lins** [comment]
I'm also having this problem, but we're using Bitrise for CI/CD

**Comment by JanithaR** [comment]
Check #391

**Comment by tpucci** [comment]
Perhaps this will help: https://github.com/luggit/react-native-config/issues/391#issuecomment-571948199

**Comment by twelve17** [comment]
https://github.com/luggit/react-native-config/issues/391#issuecomment-571948199 Didn't apply in my case (I don't see the `Build input file cannot be found: GeneratedDotEnv.m` error).  The app just always ends up with a Info.plist with empty values where the env values should be at.  Unless I run it again.  I'm going to try the other workaround in https://github.com/luggit/react-native-config/issues/391#issuecomment-566781407.

**Comment by lucianomlima** [comment]
Have the same problem with Bitrise. Defined secrets and env vars correctly there, but get `AAPT: error: resource string/FACEBOOK_APP_ID (aka com.myapp.example/FACEBOOK_APP_ID) not found`.

Add `.env` file to repository seems strange to me.

**Comment by shaddeen** [comment]
Did anyone figure this out? Same issue on Bitrise

**Comment by luis-f-lins** [comment]
@lucianomlima @shaddeen on Bitrise, I've come up with a temporary fix. I've created the following .js script and put it in the source folder. Then, in the Bitrise workflow, right after the `yarn install` step,  I added a `Script Runner` step, to run the following script with node. This fixed the problem, and I've been using it for a few months now. Hope this helps you guys!

```
const fs = require('fs');
const dotenv = require('dotenv');
const promisify = require('util').promisify;
const writeFile = promisify(fs.writeFile);
const readFile = promisify(fs.readFile);

async function main() {
  try {
    console.log('===================== THIS IS A TEMPORARY SCRIPT ===================');
    console.log('===================== UPDATING Info.plist with .env ================');
    console.log('====================================================================');

    let isProduction;
    try {
      if(fs.existsSync('./.env.production')) {
        isProduction = true;
        console.log('production environment');
      } else {
        isProduction = false;
        console.log('staging environment');
      }
    } catch (err) {
        console.error(err);
    }
    const envfile = isProduction ? await readFile('./.env.production', 'utf-8') : await readFile('./.env', 'utf-8');
    const plist = await readFile('./ios/[***  PROJECT_FOLDER  ***]/Info.plist', 'utf-8');
    const env = dotenv.parse(Buffer.from(envfile));
    let plistupdated = plist;
    Object.entries(env).forEach(([key,value]) => {
      plistupdated = plistupdated.replace(new RegExp(`\\$\\(${key}\\)`, 'g'), value);
    });
    await writeFile('./ios/[***  PROJECT_FOLDER  ***]/Info.plist', plistupdated, 'utf-8');
    await writeFile('./ios/[***  PROJECT_FOLDER  ***]/Info.plist.bak', plist, 'utf-8');
  } catch (e) {
    console.error(e);
    console.error(`Unable to read .env file`);
    process.exit(1);
  }
}

main();
```

**Comment by shaddeen** [comment]
@luis-f-lins thanks! Do you have something for android?

**Comment by luis-f-lins** [comment]
> @luis-f-lins thanks! Do you have something for android?

I don’t think this problem occurs on Android, because it’s related to the Info.plist file, which is an iOS file.

**Comment by lucianomlima** [comment]
> @lucianomlima @shaddeen on Bitrise, I've used a temporary fix based on https://github.com/luggit/react-native-config/issues/391#issuecomment-566781407. I've created the following .js script and put it in the source folder. Then, in the Bitrise workflow (I've personally changed both the staging and production workflows), right after the `yarn install` step,  I added a `Script Runner` step, to run the following script with node. This fixed the problem, and I've been using it for a few months now. Hope this helps you guys!
> 
> ```
> const fs = require('fs');
> const dotenv = require('dotenv');
> const promisify = require('util').promisify;
> const writeFile = promisify(fs.writeFile);
> const readFile = promisify(fs.readFile);
> 
> async function main() {
>   try {
>     console.log('===================== THIS IS A TEMPORARY SCRIPT ===================');
>     console.log('===================== UPDATING Info.plist with .env ================');
>     console.log('====================================================================');
> 
>     let isProduction;
>     try {
>       if(fs.existsSync('./.env.production')) {
>         isProduction = true;
>         console.log('production environment');
>       } else {
>         isProduction = false;
>         console.log('staging environment');
>       }
>     } catch (err) {
>         console.error(err);
>     }
>     const envfile = isProduction ? await readFile('./.env.production', 'utf-8') : await readFile('./.env', 'utf-8');
>     const plist = await readFile('./ios/[***  PROJECT_FOLDER  ***]/Info.plist', 'utf-8');
>     const env = dotenv.parse(Buffer.from(envfile));
>     let plistupdated = plist;
>     Object.entries(env).forEach(([key,value]) => {
>       plistupdated = plistupdated.replace(new RegExp(`\\$\\(${key}\\)`, 'g'), value);
>     });
>     await writeFile('./ios/[***  PROJECT_FOLDER  ***]/Info.plist', plistupdated, 'utf-8');
>     await writeFile('./ios/[***  PROJECT_FOLDER  ***]/Info.plist.bak', plist, 'utf-8');
>   } catch (e) {
>     console.error(e);
>     console.error(`Unable to read .env file`);
>     process.exit(1);
>   }
> }
> 
> main();
> ```

I have been resokved in a similar way, but using a bash script.

**Comment by shaddeen** [comment]
> > @luis-f-lins thanks! Do you have something for android?
> 
> I don’t think this problem occurs on Android, because it’s related to the Info.plist file, which is an iOS file.

Strange, I have the exact same issue with Android

**Comment by jaexplorer** [comment]
@luis-f-lins I'm still having issues with Bitrise, are you able to help me?

**Comment by luis-f-lins** [comment]
> @luis-f-lins I'm still having issues with Bitrise, are you able to help me?

Can you describe a little more of what you’re running into?

**Comment by jaexplorer** [comment]
Managed to figure out I was trying to fix the wrong iOS target, all good

**Comment by ryanbabbly** [comment]
I get this issue with Android using version ^1.3.3 in Microsoft App Center in my strings.xml file:

`<string moduleConfig="true" name="CodePushDeploymentKey">@string/CODE_PUSH_KEY</string>`

It builds fine locally but when I build in App Center I get this error:

```
* What went wrong:
Execution failed for task ':app:processReleaseResources'.
> A failure occurred while executing com.android.build.gradle.internal.tasks.Workers$ActionFacade
   > Android resource linking failed
     /Users/runner/work/1/s/android/app/src/main/res/values/strings.xml:5:5-92: AAPT: error: resource string/CODE_PUSH_KEY (aka babbly.app:string/CODE_PUSH_KEY) not found.
```

For now I'm patching by manually replace this value at build time, but it's not an ideal solution by any stretch of the imagination. Is there a way to fix this?

**Comment by grit96** [comment]
The issue with the first CI build not having the Info.plist processed is that the build relies on `tmp.xcconfig` existing at the start of the build (seemingly before the pre-action script is run).

My workaround is to run `touch ios/tmp.xcconfig` at the start of my CI process so that all builds will have the Info.plist processed correctly.

**Comment by rafaelmaeuer** [comment]
maybe https://github.com/luggit/react-native-config/issues/710 helps?
